docs/install/Knative-with-any-k8s.md

```diff
@@ -1,89 +1,473 @@
+   {{< tabs name="serving_networking" default="Istio" >}}
+{{% tab name="Ambassador" %}}
+
+The following commands install Ambassador and enable its Knative integration.
+
+1. Create a namespace to install Ambassador in:
+
+   ```bash
+   kubectl create namespace ambassador
+   ```
+
+1. Install Ambassador:
+
+   ```bash
+   kubectl apply --namespace ambassador \
+     --filename https://getambassador.io/yaml/ambassador/ambassador-rbac.yaml \
+     --filename https://getambassador.io/yaml/ambassador/ambassador-service.yaml
+   ```
+
+1. Give Ambassador the required permissions:
+
+   ```bash
+   kubectl patch clusterrolebinding ambassador -p '{"subjects":[{"kind": "ServiceAccount", "name": "ambassador", "namespace": "ambassador"}]}'
+   ```
+
+1. Enable Knative support in Ambassador:
+
+   ```bash
+   kubectl set env --namespace ambassador  deployments/ambassador AMBASSADOR_KNATIVE_SUPPORT=true
+   ```
+
+1. To configure Knative Serving to use Ambassador by default:
+
+    ```bash
+    kubectl patch configmap/config-network \
+      --namespace knative-serving \
+      --type merge \
+      --patch '{"data":{"ingress.class":"ambassador.ingress.networking.knative.dev"}}'
+    ```
+
+1. Fetch the External IP or CNAME:
+
+   ```bash
+   kubectl --namespace ambassador get service ambassador
+   ```
+
+   Save this for configuring DNS below.
+
+{{< /tab >}}
+
+{{% tab name="Contour" %}}
+The following commands install Contour and enable its Knative integration.
+
+1. Install the Knative Contour controller (includes Contour):
+
+   ```bash
+   kubectl apply --filename https://raw.githubusercontent.com/knative/serving/{{< version >}}/third_party/contour-latest/contour.yaml
+   ```
+
+1. To configure Knative Serving to use Contour by default:
+
+    ```bash
+    kubectl patch configmap/config-network \
+      --namespace knative-serving \
+      --type merge \
+      --patch '{"data":{"ingress.class":"contour.ingress.networking.knative.dev"}}'
+    ```
+
+1. Fetch the External IP or CNAME:
+
+   ```bash
+   kubectl --namespace projectcontour get service envoy-external
+   ```
+
+   Save this for configuring DNS below.
+
+{{< /tab >}}
+
+{{% tab name="Gloo" %}}
+_For a detailed guide on Gloo integration, see [Installing Gloo for Knative](https://docs.solo.io/gloo/latest/installation/knative/) in the Gloo documentation._
+
+The following commands install Gloo and enable its Knative integration.
+
+1. Make sure `glooctl` is installed (version 1.3.x and higher recommended):
+
+   ```bash
+   glooctl version
+   ```
+
+   If it is not installed, follow the instructions [here](https://docs.solo.io/gloo/latest/installation/knative/#install-command-line-tool-cli)
+
+1. Install Gloo and the Knative integration:
+
+    ```bash
+    glooctl install knative --install-knative=false
+    ```
+
+1. Fetch the External IP or CNAME:
+
+   ```bash
+   kubectl get svc -ngloo-system knative-external-proxy
```

Comment:
we can also provide `glooctl proxy url --name knative-external-proxy` as an alternative here